optaweb-vehicle-routing-backend/src/main/java/org/optaweb/vehiclerouting/solver/VehicleRoutingSolution.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2012 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaweb.vehiclerouting.solver;
+
+import java.text.NumberFormat;
+import java.util.List;
+
+import com.thoughtworks.xstream.annotations.XStreamAlias;
+import com.thoughtworks.xstream.annotations.XStreamConverter;
+import com.thoughtworks.xstream.annotations.XStreamInclude;
+import org.optaplanner.core.api.domain.solution.PlanningEntityCollectionProperty;
+import org.optaplanner.core.api.domain.solution.PlanningScore;
+import org.optaplanner.core.api.domain.solution.PlanningSolution;
+import org.optaplanner.core.api.domain.solution.drools.ProblemFactCollectionProperty;
+import org.optaplanner.core.api.domain.valuerange.ValueRangeProvider;
+import org.optaplanner.core.api.score.buildin.hardsoftlong.HardSoftLongScore;
+import org.optaplanner.persistence.xstream.api.score.buildin.hardsoftlong.HardSoftLongScoreXStreamConverter;
+import org.optaweb.vehiclerouting.domain.Customer;
+import org.optaweb.vehiclerouting.domain.Depot;
+import org.optaweb.vehiclerouting.domain.Location;
+import org.optaweb.vehiclerouting.domain.Vehicle;
+import org.optaweb.vehiclerouting.domain.persistable.AbstractPersistable;
+import org.optaweb.vehiclerouting.domain.timewindowed.TimeWindowedVehicleRoutingSolution;
+
+@PlanningSolution
+@XStreamAlias("VrpVehicleRoutingSolution")
+@XStreamInclude({TimeWindowedVehicleRoutingSolution.class})
+public class VehicleRoutingSolution extends AbstractPersistable {
+    protected String name;
+    private String distanceUnitOfMeasurement;
+    private List<Location> locationList;
+    private List<Depot> depotList;
+    private List<Vehicle> vehicleList;
+    private List<Customer> customerList;
+    @XStreamConverter(HardSoftLongScoreXStreamConverter.class)
```

Comment:
I see, this is the reason for depending on `optaplanner-persistence`. Not a hard requirement, but you *can* remove all `@XStream` annotations from the migrated code. The planning domain classes are not being serialized in this project.